SingularityBase/src/main/java/com/hubspot/mesos/CounterMap.java

```diff
@@ -52,12 +52,7 @@ public String toString() {
   }
 
   public void incr(K key, long amount) {
-    Counter c = map.get(key);
-
-    if (c == null) {
-      c = new Counter();
-      map.put(key, c);
-    }
+    Counter c = map.computeIfAbsent(key, k -> new Counter());
```

Comment:
small gotcha for putIfAbsent: turns out putIfAbsent returns the old value (null) so computeIfAbsent is what we want here (returns existing or computed) 
![bitmoji](https://render.bitstrips.com/v2/cpanel/9954989-211453933_4-s1-v1.png?transparent=1&palette=1&width=46)